src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -0,0 +1,644 @@
+
+  }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::deleteParticle()
+   */
+  bool deleteParticle(Particle &particle) override {
+
+    return _carrierLevel.deleteParticle(particle);
+
+  }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::deleteParticle()
+   */
+  bool deleteParticle(size_t cellIndex, size_t particleIndex) override {
+
+    return _carrierLevel.deleteParticle(cellIndex, particleIndex);
+
+  }
+
+  /**
+   * @brief Get the Hierarchy Level Of Particle object
+   * 
+   * @param p 
+   * @return Level the particle should be sorted into. 
+   */
+  unsigned int getHierarchyLevelOfParticle(const ParticleType &p) {
+
+    // If only one HG level is available all particles are sorted into this level.
+    if (_numberOfLevels == 0) {
+      return 0;
+    }
+        
+    // Get the radius of the particle in question
+    auto particleRad = p.getRad();
+
+    unsigned int level;
+
+    // Loop over all hierarchy levels
+    for (level = 0; level < _numberOfLevels; level++) {
+      // As long as the '_hierarchicalGridBorders' element is smaller than the radius continue to cycle
+      if (particleRad > _hierarchicalGridBorders[level]) break;     
+    }
+
+    if(level == 0) {
+    utils::ExceptionHandler::exception(
+      "HierarchicalGrids: Trying to get the hierarchy level of a particle which is too large to fit in the grid.\n"
+      "Particle radius {}\n"
+      "Upper limit of the largest level {}",
+      particleRad, _hierarchicalGridBorders[level]);
+    }
+    // Break and return the level
+    return level - 1;
+  }
+
+  void setHierarchicalGridBorders(std::vector<double> boundaries) {
+
+    _hierarchicalGridBorders = boundaries;
+
+  }
+
+  std::vector<double> getHierarchicalGridBorders() {
+
+    return _hierarchicalGridBorders;
+
+  }
+
+
+  private:
+
+  /**
+   * @brief Get the number Of cross-levels of the H-Grid
+   * 
+   * @param levels 
+   * @return unsigned int 
+   */
+  unsigned int getNumberOfCrossLevels(unsigned int levels) {
+
+    unsigned int numberOfCrossLevels = 0;
+    for (unsigned int i = 1; i < levels; i++)
+    {
+      numberOfCrossLevels += i;
+    }
+    
+    return numberOfCrossLevels;
+  }
+
+  /**
+  * @brief Vector containing the HGrid's different hierarchy levels of Linked Cells.
+  * 
+  */
+  std::deque<LinkedCells<Particle>> _hierarchyLevels;
+
+  /**
+   * @brief Carrier level containing all particles for functions like addParticle() etc.
+   * 
+   */
+  autopas::LinkedCells<Particle> _carrierLevel;
+
+  /**
+   * @brief Number of levels (hierarchies) in the H-Grid
+   * 
+   */
+  unsigned int _numberOfLevels;
+
+  std::array<double, 2> _particleRadiusRange;
+
+  std::array<double, 3> _boxMin; 
+  std::array<double, 3> _boxMax; 
+  double _cutoff;
+  double _skinPerTimestep; 
+  unsigned int _rebuildFrequency;
+  double _cellSizeFactor;
+  autopas::LoadEstimatorOption _loadEstimator;
```

Comment:
theoretically you could get rid of all of them too because they should also be stored in `_hierarchyLevels[0]`.